common/driver/gpu_async.c

```diff
@@ -0,0 +1,197 @@
+/**
+ *-----------------------------------------------------------------------------
+ * Title      : AXIS Gen2 Functions
+ * ----------------------------------------------------------------------------
+ * File       : axis_gen2.c
+ * Created    : 2017-02-03
+ * ----------------------------------------------------------------------------
+ * Description:
+ * Access functions for Gen2 AXIS DMA
+ * ----------------------------------------------------------------------------
+ * This file is part of the aes_stream_drivers package. It is subject to 
+ * the license terms in the LICENSE.txt file found in the top-level directory 
+ * of this distribution and at: 
+ *    https://confluence.slac.stanford.edu/display/ppareg/LICENSE.html. 
+ * No part of the aes_stream_drivers package, including this file, may be 
+ * copied, modified, propagated, or distributed except according to the terms 
+ * contained in the LICENSE.txt file.
+ * ----------------------------------------------------------------------------
+**/
+#include <gpu_async.h>
+#include <GpuAsync.h>
+#include <linux/seq_file.h>
+#include <linux/signal.h>
+#include <linux/slab.h>
+#include <nv-p2p.h>
+
+// Execute command
+void Gpu_Init(struct DmaDevice *dev, uint32_t offset) {
+   struct GpuData * gpuData;
+
+   // Setup GPU Utility
+   gpuData = (struct GpuData *)kmalloc(sizeof(struct GpuData),GFP_KERNEL);
+   dev->utilData = gpuData;
+
+   gpuData->base = dev->base + offset;
+   gpuData->writeBuffers.count = 0;
+   gpuData->readBuffers.count = 0;
+}
+
+// Execute command
+int32_t Gpu_Command(struct DmaDevice *dev, uint32_t cmd, uint64_t arg) {
+   switch (cmd) {
+
+      // Add NVIDIA Memory
+      case GPU_Add_Nvidia_Memory:
+         return(Gpu_AddNvidia(dev,arg));
+         break;
+
+      // Rem NVIDIA Memory
+      case GPU_Rem_Nvidia_Memory:
+         return(Gpu_RemNvidia(dev,arg));
+         break;
+
+      default:
+         dev_warn(dev->device,"Command: Invalid command=%i\n",cmd); 
+         return(-1);
+         break;
+   }
+}
+
+// Add NVIDIA Memory
```

Comment:
There should be refresh to what NVIDIA devices are supported (maybe point to the link that says only QUADRO and TELSA product line are supported)